### PR TITLE
PHPUnit_Util_Diff class not found fix

### LINKED ISSUE #11016: Test case fails with PHPUnit_Util_Diff class not found
This is a (multiple allowed):

* [x] bug
* [ ] enhancement
* [ ] feature-discussion (RFC)

* CakePHP Version: 2.10.
* Platform and Target: Local testing.

### What you did
Upgraded to CakePHP 2.10 which has limited support for PHPUnit 4.x and 5.x. Also added the `phpunit.phar` version `5.7.21` to `app/Vendors` to use the new version of PHPUnit.

### What happened
When I run the unit test from the browser by going to `/test.php` and my unit test fails I get the following error:

```
Fatal Error

Error: Class 'PHPUnit_Util_Diff' not found	
File: ../lib/Cake/TestSuite/Reporter/CakeHtmlReporter.php	
Line: 268
```

Class `PHPUnit_Util_Diff` seems to have been removed on PHPUnit 4.x and 5.x (it does exist and work with PHPUnit 3.7).

### What you expected to happen
The test case failure be displayed on the screen.

### PR DESCRIPTION
**Pull request by peloncano**
Added `PHPUnit 4.x+` support for the webrunner. The changes are backward compatible with PHPUnit version `3.7.28`

Fixes #11016

### CODE REVIEW

**Comment by dereuromark** [comment]
I fixed the target branch for now, master is 3.x, not 2.x.

File: lib/Cake/TestSuite/Reporter/CakeHtmlReporter.php [hunk 1]
```diff
@@ -265,7 +265,19 @@ public function paintFail($message, $test) {
...
-			echo "<br />" . $this->_htmlEntities(PHPUnit_Util_Diff::diff($expectedMsg, $actualMsg));
+			
+			$diffs = "";
+			
+			// PHPUnit 3.7.38 support
+			if(class_exists('PHPUnit_Util_Diff')) {
+				$diffs = PHPUnit_Util_Diff::diff($expectedMsg, $actualMsg);
+				
+			// PHPUnit 4.x and 5.x support
+			}else if(class_exists('SebastianBergmann\Diff\Differ')){
```
Comment (dereuromark) [inline lib/Cake/TestSuite/Reporter/CakeHtmlReporter.php]: Please respect CS

File: lib/Cake/TestSuite/Reporter/CakeHtmlReporter.php [hunk 2]
```diff
@@ -265,7 +265,15 @@ public function paintFail($message, $test) {
 		echo "<div class='msg'><pre>" . $this->_htmlEntities($message->toString());
 
 		if ((is_string($actualMsg) && is_string($expectedMsg)) || (is_array($actualMsg) && is_array($expectedMsg))) {
-			echo "<br />" . $this->_htmlEntities(PHPUnit_Util_Diff::diff($expectedMsg, $actualMsg));
+			
+			$diffs = "";
+			if (class_exists('PHPUnit_Util_Diff')) {
+				$diffs = PHPUnit_Util_Diff::diff($expectedMsg, $actualMsg);
+			} elseif (class_exists('SebastianBergmann\Diff\Differ')) {
+				$diffs = (new SebastianBergmann\Diff\Differ())->diff($expectedMsg, $actualMsg);
```
Comment (dereuromark) [inline lib/Cake/TestSuite/Reporter/CakeHtmlReporter.php]: You need to split this into 2 lines. This syntax doesnt work in PHP5.3.

**Comment by markstory** [comment]
Thank you for fixing that :+1:

**Pull request merged**